spectrocloud/kubevirt/schema/virtualmachine/spec.go

```diff
@@ -84,22 +34,120 @@ func expandVirtualMachineSpec(virtualMachine []interface{}) (kubevirtapiv1.Virtu
 	return result, nil
 }
 
-func flattenVirtualMachineSpec(in kubevirtapiv1.VirtualMachineSpec) []interface{} {
+func FlattenVirtualMachineSpec(in kubevirtapiv1.VirtualMachineSpec, resourceData *schema.ResourceData) []interface{} {
 	att := make(map[string]interface{})
 
 	if in.RunStrategy != nil {
 		att["run_strategy"] = string(*in.RunStrategy)
 	}
 	if in.Template != nil {
-		att["template"] = virtualmachineinstance.FlattenVirtualMachineInstanceTemplateSpec(*in.Template)
+		att["template"] = virtualmachineinstance.FlattenVirtualMachineInstanceTemplateSpec(*in.Template, resourceData)
 	} else {
 		att["template"] = []interface{}{} // Set to empty value
 	}
 	if in.DataVolumeTemplates != nil {
-		att["data_volume_templates"] = flattenDataVolumeTemplates(in.DataVolumeTemplates)
+		att["data_volume_templates"] = flattenDataVolumeTemplates(in.DataVolumeTemplates, resourceData)
 	} else {
 		att["data_volume_templates"] = []interface{}{} // Set to empty value
 	}
 
 	return []interface{}{att}
 }
+
+func FlattenVMMToSpectroSchema(in kubevirtapiv1.VirtualMachineSpec, resourceData *schema.ResourceData) error {
+	VM := FlattenVirtualMachineSpec(in, resourceData)[0].(map[string]interface{})
+	VMTemplate := VM["template"]
+	VMTemplateSpec := VMTemplate.([]interface{})[0].(map[string]interface{})["spec"]
+	VMTemplateSpecAttributes := VMTemplateSpec.([]interface{})[0].(map[string]interface{})
+	if err := resourceData.Set("run_strategy", VM["run_strategy"]); err != nil {
```

Comment:
need to bring it common style.

Reply:
For now added the comments and organized into mandatory and nonmandatory attributes